spec/app/domain/authentication/authn-azure/validate_application_identity_spec.rb

```diff
@@ -0,0 +1,454 @@
+    allow(system_assigned_identity_annotation).to receive(:[])
+                                                    .with(:value)
+                                                    .and_return(oid_token_field)
+
+    allow(mismatched_subscription_id_annotation).to receive(:values)
+                                                      .and_return(mismatched_subscription_id_annotation)
+    allow(mismatched_subscription_id_annotation).to receive(:[])
+                                                      .with(:name)
+                                                      .and_return("authn-azure/subscription-id")
+    allow(mismatched_subscription_id_annotation).to receive(:[])
+                                                      .with(:value)
+                                                      .and_return("mismatched-subscription-id")
+
+    allow(mismatched_resource_group_annotation).to receive(:values)
+                                                     .and_return(mismatched_resource_group_annotation)
+    allow(mismatched_resource_group_annotation).to receive(:[])
+                                                     .with(:name)
+                                                     .and_return("authn-azure/resource-group")
+    allow(mismatched_resource_group_annotation).to receive(:[])
+                                                     .with(:value)
+                                                     .and_return("mismatched-resource-group")
+
+    allow(mismatched_system_assigned_identity_annotation).to receive(:values)
+                                                               .and_return(mismatched_system_assigned_identity_annotation)
+    allow(mismatched_system_assigned_identity_annotation).to receive(:[])
+                                                               .with(:name)
+                                                               .and_return("authn-azure/system-assigned-identity")
+    allow(mismatched_system_assigned_identity_annotation).to receive(:[])
+                                                               .with(:value)
+                                                               .and_return("mismatched-system-assigned-identity")
+
+    allow(mismatched_user_assigned_identity_annotation).to receive(:values)
+                                                             .and_return(mismatched_user_assigned_identity_annotation)
+    allow(mismatched_user_assigned_identity_annotation).to receive(:[])
+                                                             .with(:name)
+                                                             .and_return("authn-azure/user-assigned-identity")
+    allow(mismatched_user_assigned_identity_annotation).to receive(:[])
+                                                             .with(:value)
+                                                             .and_return("mismatched-user-assigned-identity")
```

Comment:
Can this be extracted to a method? can make it more readable

Reply:
In other words a method that looks like the following:

```
def mock_host_constructor(host_name, host_annotation, host_value)
    allow(host_name).to receive(:values)
                                           .and_return(host_name)
    allow(host_name).to receive(:[])
                                           .with(:name)
                                           .and_return("authn-azure/host_annotation-id")
    allow(host_name).to receive(:[])
                                           .with(:value)
                                           .and_return("host_value")
end

before(:each) do
  mock_host_constructor(subscription_id_annotation, subscription-id, some-subscription-id-value)
  mock_host_constructor ...
  mock_host_constructor ...
end
```

Reply:
Yes, only it is not `host_name`. It is `host_annotation_type` or something like that.